modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -1342,214 +1314,519 @@ public GranuleLoadingResult loadRaster(
+                                    .setRenderingHints(hints)
+                                    .format(raster.getSampleModel().getDataType())
+                                    .getRenderedImage();
                 }
             }
-            // keep into account translation factors to place this tile
-            finalRaster2Model.preConcatenate((AffineTransform) mosaicWorldToGrid);
-            final Interpolation interpolation = request.getInterpolation();
-
-            // paranoiac check to avoid that JAI freaks out when computing its internal layouT on
-            // images that are too small
-            Rectangle2D finalLayout =
-                    ImageUtilities.layoutHelper(
-                            raster,
-                            (float) finalRaster2Model.getScaleX(),
-                            (float) finalRaster2Model.getScaleY(),
-                            (float) finalRaster2Model.getTranslateX(),
-                            (float) finalRaster2Model.getTranslateY(),
-                            interpolation);
-            if (finalLayout.isEmpty()) {
+        }
+        return raster;
+    }
+
+    private RenderedImage applyRescaling(
+            final RasterLayerRequest request, final Hints hints, RenderedImage raster) {
+        if (request.isRescalingEnabled()) {
+            if (noData != null && request.getReadType() == ReadType.JAI_IMAGEREAD) {
+                // Force nodata settings since JAI ImageRead may lost that
+                // We have to make sure that noData pixels won't be rescaled
+                PlanarImage t = PlanarImage.wrapRenderedImage(raster);
+                t.setProperty(NoDataContainer.GC_NODATA, noData);
+                raster = t;
+            }
+            final int[] bands = request.getBands();
+            raster = rescale(raster, hints, bands);
+        }
+        return raster;
+    }
+
+    private RasterState loadAndPrepareRaster(
+            final RequestParams params, final ReaderState readerState) throws IOException {
+
+        final RasterLayerRequest request = params.request;
+
+        RasterState state = new RasterState();
+        state.proceed = false;
+
+        final ImageReadParam readParameters = readerState.readParameters;
+        final int ovrIndex = readerState.ovrIndex;
+        final URL granuleURLUpdated = readerState.granuleURL;
+        final GranuleOverviewLevelDescriptor selectedlevel = readerState.selectedlevel;
+        final ImageReader reader = readerState.reader;
+        final boolean closeElements = false;
+        // read
+        final ReadType readType = request.getReadType();
+        state.raster =
+                readType.read(
+                        readParameters,
+                        ovrIndex,
+                        granuleURLUpdated,
+                        selectedlevel.rasterDimensions,
+                        reader,
+                        hints,
+                        closeElements);
```

Comment:
The original code was catching a Throwable around the read method. Not sure it was fully needed, just reporting that in case you want to double check.